src/schedule/ll_schedule.c

```diff
@@ -175,6 +176,17 @@ static void schedule_ll_tasks_run(void *data)
 
 	last_tick = sch->domain->last_tick;
 
+	/* last_tick is the time we asked the timer to run the LL scheduler,
+	 * let give it a quick sanity check to see if we have been blocked
+	 * for any reason. TODO: determine a tolerance - uses ~62.5uS now
+	 */
+	tick_now = platform_timer_get(timer_get());
+	if ((tick_now - last_tick) > (sch->domain->ticks_per_ms >> 4)) {
+		/* we have started late - shout about it */
+		tr_err(&ll_tr, "schedule_ll_tasks_run: started %d ticks late",
+		       (uint32_t)(tick_now - last_tick));
+	}
```

Comment:
The change is valuable for specific issue investigate but I would not qualify it as necessary  for upstream as it puts additional overhead in FW.  We are tracking delays in scheduler timer set and in case of significant delay on resume we may eventually end up with xrun on specific module which will be reported.